### PR TITLE
chore(3000): Split new sidebar out as a new feature flag

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

We didn't highlight the home navbar item (or any item for that matter) when its scene was shown. The biggest change in this PR is that now we do, like this:

### CODE REVIEW

**Comment by Twixes** [comment]
Changed the scope of this PR to be about splitting out the new sidebar-based navigation to a new `posthog-3000-nav` flag. Here's how the base `posthog-3000` works now:

**Comment by corywatilo** [comment]
[Updated nav bar icons + order](https://www.figma.com/file/uNuL5XDnOqasYuK0yfpnmV/PostHog-3000?type=design&node-id=2665%3A29308&mode=design&t=YNHeDShn7VGzVdPk-1)

(And we need a case for showing the full labels with a background and not being overlaid onto the page)

File: frontend/__snapshots__/posthog-3000-sidebar--feature-flags.png
Comment (thmsobrmlr) [inline frontend/__snapshots__/posthog-3000-sidebar--feature-flags.png:1]: The `FEATURE_FLAGS.POSTHOG_3000_NAV` needs to be added to the `with3000` Storybook decorator for this to work. I'll need to make this work for https://github.com/PostHog/posthog/pull/18123 as well, so feel free to skip.

**Review by thmsobrmlr — APPROVED** [review]
Sorry, just saw this is waiting for review. Looks great!

**Pull request merged**